### PR TITLE
 refactor: use async function

### PR DESCRIPTION
**Pull request by BaffinLee**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

1. use async function instead of generator.
2. update dependencies to lastest, support Egg 2.

##### BREAKING CHANGE:

need `node >= 8` due to async function.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 sudo: false
 language: node_js
 node_js:
-  - '6'
-  - '7'
+  - '8'
+  - '9'
```
Comment (atian25) [inline .travis.yml]: 10

File: package.json [hunk 2]
```diff
@@ -51,7 +51,7 @@
     "lib"
   ],
   "ci": {
-    "version": "6, 7",
+    "version": "8, 9",
```
Comment (atian25) [inline package.json]: 8, 10

File: .travis.yml (hunk 1 above)
Comment (BaffinLee) [reply .travis.yml]: Just `10` or `8` & `10` ?

Comment (BaffinLee) [reply .travis.yml]: @atian25

Comment (atian25) [reply .travis.yml]: don't need to modify this file manually, just update `package.json` then `node ./node_modules/egg-ci/install.js`

**Comment by dead-horse** [comment]
2.0.0

**Pull request merged**